nlprule/src/tokenizer.rs

```diff
@@ -245,75 +290,79 @@ impl Tokenizer {
 
     /// Tokenize the given sentence. This applies chunking and tagging, but does not do disambiguation.
     // NB: this is not public because it could be easily misused by passing a text instead of one sentence.
-    pub(crate) fn tokenize<'t>(&'t self, sentence: &'t str) -> Vec<IncompleteToken<'t>> {
-        let mut current_char = 0;
+    pub(crate) fn tokenize<'t>(&'t self, sentence: &'t str) -> Option<IncompleteSentence<'t>> {
+        if sentence.trim().is_empty() {
+            return None;
+        }
+
         let token_strs = self.get_token_strs(sentence);
+
         let mut tokens: Vec<_> = token_strs
             .iter()
             .enumerate()
-            .map(|(i, x)| {
-                let char_start = current_char;
-                let ptr = x.as_ptr() as usize;
-                current_char += x.chars().count();
+            .filter(|(_, token_text)| !token_text.trim().is_empty())
+            .map(|(i, token_text)| {
+                let byte_start = token_text.as_ptr() as usize - sentence.as_ptr() as usize;
+                let char_start = sentence[..byte_start].chars().count();
```

Comment:
I am sure, there is a faster way than to recount from the beginning :upside_down_face:

Reply:
The problem here is the `.filter` before the `.map` so we'd loose some characters. It would be possible to calculate char length -> filter -> map but that's not worth it imo for surely unmeasurable performance difference.

Reply:
I'll add a comment here to clarify that.